CMakeLists.txt

```diff
@@ -1,6 +1,6 @@
 cmake_minimum_required(VERSION 2.6)
 
-project(Effective_Modern_C++ CXX)
+project(Effective_Modern_C++ C CXX)
```

Comment:
It's for (cmake) pthread compatibility.
Adding pthread like i do include some tests files (intern codes) to check if pthread is accessible for the current system.
It's a CMake mechanism (througth 'FindThreads.cmake').

MMmMMMM after reading the file 'FindThreads.cmake' (https://github.com/Kitware/CMake/blob/master/Modules/FindThreads.cmake) i'm not sure it's necessary after all ... i will check tomorrow to remove this line.